packages/@react-stately/steplist/src/useStepListState.ts

```diff
@@ -38,25 +38,29 @@ export interface StepListState<T> extends SingleSelectListState<T> {
 export function useStepListState<T extends object>(props: StepListProps<T>): StepListState<T> {
   let state = useSingleSelectListState<T>(props);
 
-  let [lastCompletedStep, setLastCompletedStep] = useControlledState<Key>(props.lastCompletedStep, props.defaultLastCompletedStep, props.onLastCompletedStepChange);
+  let [lastCompletedStep, setLastCompletedStep] = useControlledState<Key | null>(props.lastCompletedStep, props.defaultLastCompletedStep ?? null, props.onLastCompletedStepChange);
   const {setSelectedKey: realSetSelectedKey, selectedKey, collection} = state;
   const {indexMap, keysLinkedList} = useMemo(() => buildKeysMaps(collection), [collection]);
   const selectedIdx = indexMap.get(selectedKey);
 
-  const isCompleted = useCallback((step: Key) => {
+  const isCompleted = useCallback((step: Key | null | undefined) => {
     if (step === undefined) {
       return false;
     }
 
-    return indexMap.get(step) <= indexMap.get(lastCompletedStep);
+    return (step !== null &&
+      lastCompletedStep !== null &&
+      indexMap.has(step) &&
+      indexMap.has(lastCompletedStep) &&
+      indexMap.get(step!)! <= indexMap.get(lastCompletedStep!)!);
```

Comment:
It's annoying that after all the checks you still need to assert it won't be null for typescript.

Could line 47 and 48 be merged into this as just another check? Or could we make this simpler by moving some of the not null assertions up there.

Reply:
got a little overly exuberant, i can remove the not null assertions for inside the 'get'